easybuild/framework/easyblock.py

```diff
@@ -3900,6 +3955,14 @@ def xs2str(xs):
         else:
             self.log.debug("Skipping RPATH sanity check")
 
+        if get_software_root('CUDA'):
```

Comment:
@boegel We have an EESSI-specific complication here. We drop CUDA to a build time dep so that we don't depend on the CUDA module at runtime. This means that we won't execute this code path so we need to trigger the module load here.

Reply:
I think you're right, but just to double check: are the build dependencies unloaded at sanity check time?

Could we fix this through an EasyBuild hook in EESSI, that loads the CUDA that was a build dependency also in the sanity_check_step (and unloads after)? Should also work for `EESSI-extend`, and no changes on the framework side needed...

Reply:
Wait... actually, I don't think you're right. Because I just did this with `EESSI-extend`, and it _did_ run the CUDA sanity check...? I'm not sure why, I would have _expected_ the problem you mentioned. So... why didn't it appear?

Reply:
I wonder what happens in the `--module-only` case, when the sanity check step is being run without building first? Perhaps this really is expected behaviour?

Reply:
matching the `rpath` sanity check behaviour makes sense to me